internal/xds/rbac/rbac_engine_test.go

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package rbac
+
+import (
+	"testing"
+
+	v3rbacpb "github.com/envoyproxy/go-control-plane/envoy/config/rbac/v3"
+	v3routepb "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+	v3matcherpb "github.com/envoyproxy/go-control-plane/envoy/type/matcher/v3"
+	"google.golang.org/grpc/internal/grpctest"
+)
+
+type s struct {
+	grpctest.Tester
+}
+
+func Test(t *testing.T) {
+	grpctest.RunSubTests(t, s{})
+}
+
+// TestRBACEngine tests the RBAC Engine by configuring the engine in different
```

Comment:
This does not test all possible matchers with all possible combinations (you would have caught the issue with the ParseCIDR function if you had tests for that).

You have a few options:
- Break this into multiple tests
  - one test for `NewEngine` which passes different `v3rbacpb.RBAC` protos and verifies if the function errors or not
  - one test for `Evaluate` which constructs a gigantic policy and tests with different values for RPC args
    - this should test or, and, not and every other possible case
- Or just do it here like you are currently doing, and cover all cases

I find it useful when writing PRs like this to check the test coverage to ensure that I am covering every possible case. https://blog.golang.org/cover.

Reply:
Added many tests. Test line coverage is now 88.5%.